File: frontend/types/ingestion.ts
export type TMDbType = "movie" | "tv" | "person";

export interface TMDbIngestionRequest {
  tmdb_id: number;
  tmdb_type: TMDbType;
  import_media: boolean;
}

export interface TMDbSearchRequest {
  query: string;
}

export interface YouTubeIngestionRequest {
  entity_id: number;
  query?: string | null;
  max_results: number;
}

export interface IngestionJob {
  id: number;
  source_name: string;
  job_type: string;
  status: string;
  message: string | null;
  created_at: string;
  updated_at: string;
}

export interface IngestionResult {
  job_id: number;
  status: string;
  message: string;
  entity_id: number | null;
  created_entity: boolean;
  updated_entity: boolean;
}

export interface TMDbSearchResult {
  id: number;
  media_type?: string;
  title?: string;
  name?: string;
  overview?: string;
  release_date?: string;
  first_air_date?: string;
}

export interface TMDbSearchResponse extends IngestionResult {
  results?: TMDbSearchResult[];
}

File: backend/app/schemas/ingestion.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field, field_validator


TMDbType = Literal["movie", "tv", "person"]


class TMDbIngestionRequest(BaseModel):
    tmdb_id: int = Field(..., gt=0)
    tmdb_type: TMDbType
    import_media: bool = False


class TMDbSearchRequest(BaseModel):
    query: str = Field(..., min_length=1)

    @field_validator("query")
    @classmethod
    def query_must_not_be_blank(cls, value: str) -> str:
        stripped = value.strip()
        if not stripped:
            raise ValueError("Query is required.")
        return stripped


class YouTubeIngestionRequest(BaseModel):
    entity_id: int = Field(..., gt=0)
    query: str | None = None
    max_results: int = Field(default=5, ge=1, le=25)

    @field_validator("query")
    @classmethod
    def clean_query(cls, value: str | None) -> str | None:
        if value is None:
            return None
        stripped = value.strip()
        return stripped or None


class IngestionJobRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    source_name: str
    job_type: str
    status: str
    message: str | None
    created_at: datetime
    updated_at: datetime


class IngestionResult(BaseModel):
    job_id: int
    status: str
    message: str
    entity_id: int | None = None
    created_entity: bool = False
    updated_entity: bool = False
